crates/node/src/lib.rs

```diff
@@ -603,6 +598,12 @@ async fn provision_cluster_metadata(
     .await
     .context("failed provisioning the initial logs")?;
 
+    let result = retry_on_network_error(common_opts.network_error_retry_policy.clone(), || {
+        metadata_store_client.provision(&initial_nodes_configuration)
+    })
+    .await
+    .context("failed provisioning the initial nodes configuration")?;
+
```

Comment:
In principle, the thinking here makes sense but I'm not sure if this will actually work with the new metadata store. If the metadata store is stuck in provisioning step, we won't be able to write the partition_table and logs metadata above to start with.

Reply:
Yes, this won't unfortunately work as Ahmed pointed out.